### DISCUSSION TITLE
Hash gives Underlying HTTP server died with "null code"

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#3672] [Help] [answered]
Good morning
I have this weird problem 
When I am trying to use `await Hash.make()` method the server dies with this warning

```
[ warn ]  Underlying HTTP server died with "null code"
```
and in Postman I get `Bad Gateway` response
if I comment the `Hash.make` line everything works as expected

I have no clue how to fix this, am I doing something wrong?

**Comment by RomainLanz** [comment]
Hey @melokki! 👋🏻 

Do you have anything else in your console?
It should print an error if you are in `development` mode.

Note that the `Hash` module [requires you to install a hasher](https://docs.adonisjs.com/guides/security/hashing#available-hashers).

```sh
# If using bcrypt
npm i phc-bcrypt

# If using argon2
npm i phc-argon2
```

**Reply by melokki** [reply]
No, I don't have anything else in the console, just that warning message and that's it.
The hasher is installed, I'm using the `phc-argon2`

**Comment by melokki** [comment]
What I've found is that, when I run the application from the local environment, the hashing works just fine, but when I run it from a docker container it throws me the above error. Don't know how to debug it further :(
any suggestions?

**Reply by thetutlage** [reply]
Can you share the complete screenshot of your terminal?

**Reply by melokki** [reply]
this is how it looks 

![image](https://user-images.githubusercontent.com/5610232/163304692-46cfefb4-e349-431e-888b-bf7bd6b9663a.png)

**Reply by melokki** [reply]
I also tried to generate a hash from a repl session inside the docker container and I got the following output

![image](https://user-images.githubusercontent.com/5610232/163819727-cb97996c-7074-4368-99ad-66f3d68f04ec.png)

and I am getting the same error when I am trying to authenticate a user, don't know what else to do

**Comment by melokki** [comment, marked as answer]
My latest finding is this:
before I was using `node:17-alpine` image for my container. I changed the image to `node:17` and now works as expected.

What could possibly cause this strange behavior on the alpine container?
